lib/rubocop/cop/style/for.rb

```diff
@@ -99,14 +99,39 @@ def incorrect_style_detected(node)
 
         def autocorrect_to_each(node)
           item, enumerable = deconstruct_for(node)
-          loc_begin = node.loc.begin
-          end_pos = loc_begin ? loc_begin.end_pos : enumerable.loc.end.end_pos
+
+          end_pos = end_position(node, enumerable)
+
           replacement_range = replacement_range(node, end_pos)
-          correction = "#{enumerable.source}.each do |#{item.source}|"
 
+          enum_source = enumerable_source(enumerable)
+
+          correction = "#{enum_source}.each do |#{item.source}|"
           ->(corrector) { corrector.replace(replacement_range, correction) }
         end
 
+        def end_position(node, enumerable)
+          if node.do?
+            node.loc.begin.end_pos
+          elsif enumerable.begin_type?
+            enumerable.loc.end.end_pos
+          else
+            enumerable.loc.expression.end.end_pos
+          end
+        end
+
+        def enumerable_source(enumerable)
+          return "(#{enumerable.source})" if wrap_into_parentheses?(enumerable)
+
+          enumerable.source
+        end
+
+        def wrap_into_parentheses?(enumerable)
+          !enumerable.begin_type? &&
```

Comment:
This is needed for the case:
```
for n in 1...value
....
```

when range is not enclosed in parenthesis the `begin_type?`  check returns false. Then we know that autocorrect must wrap the enumerable.source into `( ... )`. Otherwise it would generate invalid code:
`1...value.each`

Reply:
Thanks for the explanation. I have one concern.
Even without this condition all test cases will succeed.

```diff
% g diff
diff --git a/lib/rubocop/cop/style/for.rb b/lib/rubocop/cop/style/for.rb
index 2c4a5e197..a14bf21c3 100644
--- a/lib/rubocop/cop/style/for.rb
+++ b/lib/rubocop/cop/style/for.rb
@@ -127,7 +127,6 @@ module RuboCop
         end

         def wrap_into_parentheses?(enumerable)
-          !enumerable.begin_type? &&
-            (enumerable.irange_type? ||
-             enumerable.erange_type?)
+          enumerable.irange_type? ||
+            enumerable.erange_type?
         end
```

```console
% bundle exec rspec spec/rubocop/cop/style/for_spec.rb
Run options: include {:focus=>true}

All examples were filtered out; ignoring {:focus=>true}

Randomized with seed 12381
.....................

Finished in 0.12819 seconds (files took 1.32 seconds to load)
21 examples, 0 failures

Randomized with seed 12381
```

Could you add a new test case that requires that condition?

Reply:
I think that's enough because there are written test cases with and without parentheses.
Thank you!